src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.Snapshotter.cs

```diff
@@ -0,0 +1,250 @@
+                // the last time we captured them.
+                // NullableWalker.variableBySlot is allocated in an exponential fashion: every time
+                // a new slot is created that's off the end of the array, the array size is doubled.
+                // Therefore, we only want to take the allocated identifiers, and verify that any slots
+                // after the ones that we have are empty in Debug to verify. Slot 0 is always allocated,
+                // so if _lastVariableBySlot is empty then we've never cached anything and need to cache
+                if (_walkerToSnapshot._variableSlot.Count != _lastVariableSlot.Count
+                    || _lastVariableBySlot.Length == 0)
+                {
+                    var variableBySlotBuilder = ArrayBuilder<VariableIdentifier>.GetInstance(_walkerToSnapshot.variableBySlot.Length);
+                    var walkerVariableBySlot = _walkerToSnapshot.variableBySlot;
+                    int i;
+                    for (i = 0; i < walkerVariableBySlot.Length; i++)
+                    {
+                        // The first slot always does not exist for the purposes of caching and can be ignored
+                        // in the exit condition check
+                        if (i != 0 && !walkerVariableBySlot[i].Exists)
+                        {
+                            break;
+                        }
+
+                        variableBySlotBuilder.Add(walkerVariableBySlot[i]);
+                    }
+
+#if DEBUG
+                    verifyRemainderNonExistant(walkerVariableBySlot, i);
+#endif
+
+                    _lastVariableBySlot = variableBySlotBuilder.ToImmutableAndFree();
+                    _lastVariableSlot = _walkerToSnapshot._variableSlot.ToImmutableDictionary();
+                }
+#if DEBUG
+                else
+                {
+                    // In debug mode, we verify the contents of the arrays to ensure that the invariant
+                    // is maintained. Because variableBySlot is allocated exponentially, we can only
+                    // verify that the length of our cached version is equal or less than the source
+                    // array, and then verify that anything we haven't cached does not exist.
+                    Debug.Assert(_lastVariableBySlot.Length <= _walkerToSnapshot.variableBySlot.Length);
+                    int i;
+                    for (i = 0; i < _lastVariableBySlot.Length; i++)
+                    {
+                        Debug.Assert(compareIdentifiers(_lastVariableBySlot[i], _walkerToSnapshot.variableBySlot[i]));
+                    }
+                    verifyRemainderNonExistant(_walkerToSnapshot.variableBySlot, i);
+
+                    Debug.Assert(_lastVariableSlot.Count == _walkerToSnapshot._variableSlot.Count);
+                    foreach ((VariableIdentifier key, int value) in _lastVariableSlot)
+                    {
+                        if (!key.Exists) continue;
```

Comment:
>continue; [](start = 41, length = 9)

Is this code reachable?

Reply:
Yes, slot 0 never exists. However, this code is now gone.

---
In reply to: [288758660](https://github.com/dotnet/roslyn/pull/35850#discussion_r288758660) [](ancestors = 288758660)